x/market/types/message_deal.go

```diff
@@ -71,3 +71,44 @@ func (msg *MsgCreateDeal) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{creator}
 }
+
+var _ sdk.Msg = &MsgSellData{}
+
+func NewSellData(cert DataValidationCertificate, seller string) *MsgSellData {
+	return &MsgSellData{
+		Cert:   &cert,
+		Seller: seller,
+	}
+}
+
+func (msg *MsgSellData) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgSellData) Type() string {
+	return "SellData"
+}
+
+// ValidateBasic is validation for MsgSellData.
+// TODO: Basic Validation for MsgSellData.
+func (msg *MsgSellData) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Seller)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+
+	return nil
+}
+
+func (msg *MsgSellData) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgSellData) GetSigners() []sdk.AccAddress {
+	signer, err := sdk.AccAddressFromBech32(msg.Cert.UnsignedCert.GetRequesterAddress())
```

Comment:
There is nothing wrong here, but I think the signer should be obtained from `msg.Seller`.
Strictly speaking, the `requesterAddress` is set by data validator in process of data validation, so it is not the signer of this transaction.
To sell data successfully, the seller and requesterAddress (in data certificate) should be the same, but I think this check should be done in validation of tx.

If I miss something in this review (including the next [pr](https://github.com/medibloc/panacea-core/pull/260), please let me know. Thank you!

Reply:
Good point. I also think we should use `msg.Seller`.

Reply:
Okay Thank you! I'll change it too.